docs/rules/prefer-presence-queries.md

```diff
@@ -0,0 +1,63 @@
+# Enforce specific queries when checking appearance or disappearance of elements (prefer-presence-queries)
+
+The (DOM) Testing Library allows to query DOM elements using different types of queries such as `getBy*` and `queryBy*`. Using `getBy*` throws an error in case the element is not found, while `queryBy` returns null instead of throwing. These differences are useful in some situations:
+
+- using `getBy*` queries when asserting if element is present, so if the element is not found the error thrown will offer better info than asserting with other queries which not throw.
+- using `queryBy*` queries when asserting if element is not present, so the test doesn't fail immediately when the element it's not found and the assertion can be executed properly.
+
+## Rule details
+
+This rule fires whenever:
+
+- `queryBy*` is used to assert element **is** present with `.toBeInTheDocument()`, `toBeTruthy()` or `.toBeDefined()` matchers or negated matchers from following case.
+- `getBy*` is used to assert element **is not** present with `.toBeNull()` or `.toBeFalsy()` matchers or negated matchers from previous case.
+
+Examples of **incorrect** code for this rule:
+
+```js
+test('some test', () => {
+  render(<App />);
+
+  // check element is present with `queryBy*`
+  expect(screen.queryByText('button')).toBeInTheDocument();
+  expect(screen.queryByText('button')).toBeTruthy();
+  expect(screen.queryByText('button')).toBeNull();
+  expect(screen.queryByText('button')).not.toBeNull();
+  expect(screen.queryByText('button')).not.toBeFalsy();
+
+  // check element is NOT present with `getBy*`
+  expect(screen.getByText('loading')).not.toBeInTheDocument();
+  expect(screen.getByText('loading')).not.toBeTruthy();
+  expect(screen.getByText('loading')).not.toBeNull();
+  expect(screen.getByText('loading')).toBeNull();
+  expect(screen.getByText('loading')).toBeFalsy();
```

Comment:
i'm fine w/ these but less concerned about them since the tests will actually fail if they are present

Reply:
I'm unsure if these must be included in this rule.

Newcomers to TL, might not know that these queries throw; by writing tests like this it makes these explicit and easier to understand.

I've seen `getBy` queries get removed because they didn't know they fail underneath.

Reply:
(also, not against it) just wanted to share my experience :)

Reply:
@timdeschryver yeah i was on the fence initially as the tests themselves would catch that, but i figured A) they're not hurting B) when used w/ the eslint plugin for IDEs it can be a quick check before you run the test...

Reply:
I don't know if I'm following. The problem is reporting `getBy*` for asserting element absence? Or reporting `toBeFalsy` matcher?

@timdeschryver I'd recommend go through the related issue for original discussion.

Reply:
And this rule won't be enabled by default, so it's not that intrusive imho.

Reply:
Again, sorry for the confusion - these does add value.
The sentence "I'm unsure if these must be included in this rule." was a leftover...